BaikalDataGenerator/Source/render.cpp

```diff
@@ -180,32 +180,32 @@ Render::Render(const std::filesystem::path& scene_file,
     }
 }
 
-void Render::UpdateCameraSettings(CameraIterator cam_state)
+void Render::UpdateCameraSettings(const CameraInfo& cam_state)
 {
-    if (cam_state->aperture != m_camera->GetAperture())
+    if (cam_state.aperture != m_camera->GetAperture())
     {
-        m_camera->SetAperture(cam_state->aperture);
+        m_camera->SetAperture(cam_state.aperture);
     }
 
-    if (cam_state->focal_length != m_camera->GetFocalLength())
+    if (cam_state.focal_length != m_camera->GetFocalLength())
     {
-        m_camera->SetFocalLength(cam_state->focal_length);
+        m_camera->SetFocalLength(cam_state.focal_length);
     }
 
-    if (cam_state->focus_distance != m_camera->GetFocusDistance())
+    if (cam_state.focus_distance != m_camera->GetFocusDistance())
     {
-        m_camera->SetFocusDistance(cam_state->focus_distance);
+        m_camera->SetFocusDistance(cam_state.focus_distance);
     }
 
     auto cur_pos = m_camera->GetPosition();
     auto at = m_camera->GetForwardVector();
     auto up = m_camera->GetUpVector();
 
-    if (!RoughCompare(cur_pos, cam_state->pos) ||
-        !RoughCompare(at, cam_state->at) ||
-        !RoughCompare(up, cam_state->up))
+    if (!RoughCompare(cur_pos, cam_state.pos) ||
```

Comment:
A camera setup happens every 30 minutes. The optimization does not make any sense.

Reply:
Nope